tests/unit/amo/components/TestLanguageTools.js

```diff
@@ -0,0 +1,152 @@
+    i18n = fakeI18n(),
+    store = dispatchClientMetadata().store,
+    ...props
+  } = {}) {
+    return shallowUntilTarget(
+      <LanguageTools i18n={i18n} store={store} {...props} />,
+      LanguageToolsBase
+    );
+  }
+
+  it('renders LoadingText if addons are not set', () => {
+    const root = renderShallow();
+
+    expect(root.find(LoadingText)).not.toHaveLength(0);
+  });
+
+  it('renders LoadingText if addons are empty', () => {
+    const { store } = dispatchClientMetadata();
+    store.dispatch(loadAddonResults({ addons: {} }));
+    const root = renderShallow({ store });
+
+    expect(root.find(LoadingText)).not.toHaveLength(0);
+  });
+
+  it('renders language tools in your locale', () => {
+    const { store } = dispatchClientMetadata({ lang: 'fr' });
+    store.dispatch(loadAddonResults({ addons }));
+    const root = renderShallow({ store });
+
+    const dictionary = root.find('.LanguageTools-in-your-locale-list-item')
+      .at(0);
+    const langPack = root.find('.LanguageTools-in-your-locale-list-item')
+      .at(1);
+
+    expect(root.find(LoadingText)).toHaveLength(0);
+    expect(root.find('.LanguageTools-in-your-locale')).toHaveLength(1);
+    expect(dictionary).toHaveLength(1);
+    expect(dictionary.find(Link))
+      .toHaveProp('children', 'le French Dictionary');
+    expect(langPack).toHaveLength(1);
+    expect(langPack.find(Link))
+      .toHaveProp('children', 'French Language Pack');
+  });
+
+  it('omits "language tools in your locale" section if none available', () => {
+    const { store } = dispatchClientMetadata({ lang: 'pt-BR' });
+    store.dispatch(loadAddonResults({ addons }));
+    const root = renderShallow({ store });
+
+    expect(root.find('.LanguageTools-in-your-locale')).toHaveLength(0);
+  });
+
+  it('renders language packs in the table view for the right language', () => {
+    const { store } = dispatchClientMetadata();
+    store.dispatch(loadAddonResults({ addons }));
+    const root = renderShallow({ store });
+
+    expect(root.find('.LanguageTools-lang-en-GB')).toHaveLength(1);
+    expect(root.find('.LanguageTools-lang-en-US')).toHaveLength(1);
+    expect(root.find('.LanguageTools-lang-fr')).toHaveLength(1);
+    expect(root.find('.LanguageTools-lang-ur')).toHaveLength(1);
+    expect(root.find('.LanguageTools-lang-zh-TW')).toHaveLength(1);
+  });
+
+  it('renders multiple addons in a list using LanguageToolList', () => {
+    const { store } = dispatchClientMetadata();
+    store.dispatch(loadAddonResults({ addons }));
+    const root = renderShallow({ store });
+
+    const dictionaryList = root
+      .find('.LanguageTools-lang-en-GB-dictionaries')
+      .find(LanguageToolList);
+    const languagePackList = root
+      .find('.LanguageTools-lang-en-GB-languagePacks')
+      .find(LanguageToolList);
+    expect(dictionaryList).toHaveLength(1);
+    expect(languagePackList).toHaveLength(1);
+    expect(dictionaryList.shallow().find('li')).toHaveLength(1);
```

Comment:
I think we should avoid this sort of thing -- the test is reaching into a separate component. You should be able to it in two passes:

1. Make sure `LanguageTools` is passing the right props to `LanguageToolList`
2. Assert that `LanguageToolList` is rendering the right thing.